### PR TITLE
Customer Name for Due Sales & System info changes

### PR DESCRIPTION
**Pull request by WebShells**
Customer Name will be required in order to continue the sales once due payment is selected as per #2397.

### CODE REVIEW

File: application/views/sales/register.php [hunk 1]
```diff
@@ -473,15 +473,25 @@
 							</tr>
 						</table>
 					<?php echo form_close(); ?>
-						<?php
-						// Only show this part if the payment cover the total and in sale or return mode
-						if($pos_mode == '1')
+					<?php
```
Comment (daN4cat) [inline application/views/sales/register.php]: There are indentation issues, please fix.

File: application/language/en-US/sales_lang.php [hunk 2]
```diff
@@ -174,3 +174,4 @@
 $lang["sales_work_order_number_duplicate"] = "Work Order Number must be unique.";
 $lang["sales_work_order_sent"] = "Work Order sent to";
 $lang["sales_work_order_unsent"] = "Work Order failed to be sent to";
+$lang["sales_customer_optional"] = "(Required for Due Payments)";
```
Comment (daN4cat) [inline application/language/en-US/sales_lang.php]: Could you please add en_GB translation too?

File: application/views/sales/register.php [hunk 3]
```diff
@@ -474,14 +474,24 @@
 						</table>
 					<?php echo form_close(); ?>
 						<?php
+						$payment_type = $this->input->post('payment_type');							
 						// Only show this part if the payment cover the total and in sale or return mode
-						if($pos_mode == '1')
-						{
+
+						if($pos_mode == '1' && $payment_type != $this->lang->line('sales_due') && !isset($customer))
+							{
```
Comment (daN4cat) [inline application/views/sales/register.php]: Indentation issues

File: application/views/sales/register.php [hunk 4]
```diff
@@ -474,14 +474,24 @@
...
 						?>
 						<div class='btn btn-sm btn-success pull-right' id='finish_sale_button' tabindex="<?php echo ++$tabindex; ?>"><span class="glyphicon glyphicon-ok">&nbsp</span><?php echo $this->lang->line('sales_complete_sale'); ?></div>
 						<?php
 						}
 						?>
+						<?php							
+						if($pos_mode == '1' && $payment_type = $this->lang->line('sales_due') && isset($customer))
```
Comment (jekkos) [inline application/views/sales/register.php]: wonder what the best thing to do is here. Should we disable the button or just not show it at all? I hope the layout doesn't change too much when removing this div?

**Comment by odiea** [comment]
Sale when due is added without a customer.
2019-03-14 15_40_15-Window
After due is added with a customer
2019-03-14 15_40_56-Window

File: application/views/sales/register.php (hunk 4 above)
Comment (WebShells) [reply application/views/sales/register.php]: I see its fine... As you can see below in @odiea screenshots, not a big change ...

**Comment by WebShells** [comment]
Copy Content

@odiea @objecttothis ? Copy info will copy table including html Format to be pasted in Git Issue...

**Comment by odiea** [comment]
I tried your new updated version but I get the same copy paste results as the earlier version. Nothing like what you are displaying in this image.

**Comment by WebShells** [comment]
@odiea the latest comment i showed isn't yet updated on the code, its local screenshot... So if its fine ill re-commit once editions are made...

**Comment by odiea** [comment]
Ok Thanks. And thanks for submitting the payment due issue.

**Comment by WebShells** [comment]
Anytime 🙂

**Comment by jekkos** [comment]
Don't know if feasible but would be nice to [have a copy to clipboard function ](https://hackernoon.com/copying-text-to-clipboard-with-javascript-df4d4988697f)

**Comment by objecttothis** [comment]
@wshells it looks good. I don't think the GitHub issue form will accept HTML and format it properly. I recommend making the copy to clipboard version a simple text version

**Comment by WebShells** [comment]
@jekkos already i added js in order to copy table to clipboard, ill re-update @objecttothis i meant the text alignment not html format :x

**Comment by WebShells** [comment]
@odiea can you try it now please ?

**Comment by odiea** [comment]
I tried it so you can see the issue where I pasted it. Looks good.

**Comment by objecttothis** [comment]
Looks good.  The DB version is still reporting the client version when it should be reporting the DB server version.

**Comment by objecttothis** [comment]
Also it looks like from @odiea's screenshot that the hash is reporting as $id

**Comment by odiea** [comment]
Yes it is because I am using my dev version not the bintray version.

**Comment by objecttothis** [comment]
Ok nevermind then.

**Comment by odiea** [comment]
Posted this new system info copy and paste. Line breaks look good

**Comment by daN4cat** [comment]
Ready for a merge?

**Comment by daN4cat** [comment]
I guess we would need to change the bug report template to adjust to the "go there and copy and paste the stuff here"

**Comment by WebShells** [comment]
@daN4cat yes we should modify the issue template to direct people to copy info, but i'd like first to find a solution for the check list we have before the system info. Ex: Read opened/closed issues, checked FAQ, etc... Maybe i can find a way to add them to the page in a static way...
We can merge what we currently have, until i check other editions...

**Comment by daN4cat** [comment]
Merged

**Pull request merged**